src/util.h

```diff
@@ -114,6 +137,10 @@ float smooth(float x) attr_const;
 float smoothreclamp(float x, float old_min, float old_max, float new_min, float new_max) attr_const;
 float sanitize_scale(float scale) attr_const;
 
+bool point_in_ellipse(complex p, Ellipse e) attr_const;
+double lineseg_circle_intersect(LineSegment seg, Circle c) attr_const;
+bool lineseg_ellipse_intersect(LineSegment seg, Ellipse e) attr_const;
+
```

Comment:
maybe a collision.c is better?

Reply:
I do plan to refactor/split up `utils.c` soon™, and I agree these should get their own file. I just didn't want to conflate too many things in this pull request, so I put them there temporarily. I think I'd prefer `geometry.c` or something like that, though.